parsers/src/main/scala/uk/gov/ons/addressIndex/crfscala/CrfAggregateFeature.scala

```diff
@@ -0,0 +1,24 @@
+package uk.gov.ons.addressIndex.crfscala
+
+import uk.gov.ons.addressIndex.crfscala.CrfAggregateFeatureAnalyser.CrfAggregateFeatureAnalyser
+import uk.gov.ons.addressIndex.crfscala.CrfScala.{CrfToken, CrfTokens}
+
+//todo scaladoc
+trait CrfAggregateFeature[T] {
+
+  /**
+    * @return a function which returns an instance of T
+    */
+  def analyser(): CrfAggregateFeatureAnalyser[T]
+
+  /**
+    * @return name
+    */
+  def name(): String
```

Comment:
`()` is by consensus an indicator of side effects, if there are none you should prefer the simple variant: `def name: String`.